File: HealthcareSystem.Application/Requests/AppointmentRequests.cs
namespace HealthcareSystem.Application.Requests;

public static class AppointmentRequests {
    public record GetAppointmentsRequest(
        Guid? DoctorId, string? UserId, int? PageIndex, int? PageSize,
        DateTimeOffset? StartTime, DateTimeOffset? EndTime
    );

    public record CreateAppointmentRequest(Guid ScheduleId, string UserEmail);
}

File: HealthcareSystem.Client/src/app/interfaces/requests/appointmentRequests.ts
export interface GetAppointmentsRequest {
  doctorId?: string;
  userId?: string;
  pageIndex?: number;
  pageSize?: number;
  startTime?: Date;
  endTime?: Date;
}

export interface CreateAppointmentRequest {
  scheduleId: string;
  userEmail: string;
}
